### PR TITLE
clipse: init at 0.0.6

### PR DESCRIPTION
**Pull request by savedra1**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/cl/clipse/package.nix
Comment (afh) [inline pkgs/by-name/cl/clipse/package.nix:3]: Allow re-use of variables…
```suggestion
buildGoModule rec {
```

File: pkgs/by-name/cl/clipse/package.nix
Comment (afh) [inline pkgs/by-name/cl/clipse/package.nix:10]: … re-use `version`
```suggestion
    rev = "v${version}";
```

File: pkgs/by-name/cl/clipse/package.nix
Comment (afh) [inline pkgs/by-name/cl/clipse/package.nix:17]:
```suggestion
    description = "A useful clipboard manager TUI for Unix";
```

File: pkgs/by-name/cl/clipse/package.nix
Comment (afh) [inline pkgs/by-name/cl/clipse/package.nix:21]: Please add yourself to [`maintainers/maintainers-list.nix`](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix) in a separate commit on this PR; see [example PR](https://github.com/NixOS/nixpkgs/pull/273272/commits/0feca8a9559ca4c9372e3a7c2f1c77898e077595) and [Commit conventions](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions) :  
> When adding yourself as maintainer in the same pull request, make a separate commit with the message maintainers: add <handle>. Add the commit before those making changes to the package or module. See [Nixpkgs Maintainers](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) for details.

File: pkgs/by-name/cl/clipse/package.nix
Comment (afh) [inline pkgs/by-name/cl/clipse/package.nix:20]: The description says that clipse is "for Unix" yet the platforms only include `linux`. Can clipse be used under darwin? If yes then `platforms` should include it, if not then the `description` should be changed as it is misleading.

File: pkgs/by-name/cl/clipse/package.nix
Comment (afh) [inline pkgs/by-name/cl/clipse/package.nix]: The following formatting is quite popular with nixpkgs:
```suggestion
{ lib 
, buildGoModule
, fetchFromGitHub 
```

**Review by afh — CHANGES_REQUESTED** [review]
Thank you for this PR, @savedra1. Please find below a few comments from a first glance… Hope this helps 🙂

File: pkgs/by-name/cl/clipse/package.nix
Comment (afh) [reply pkgs/by-name/cl/clipse/package.nix:20]: ℹ️ I was able to build and run clipse locally on `aarch64-darwin` 🎉 

```suggestion
    platforms = platforms.unix;
```

File: pkgs/by-name/cl/clipse/package.nix
Comment (savedra1) [reply pkgs/by-name/cl/clipse/package.nix]: Thank you for your help @afh :pray: 

I've made the suggested changes now - please let me know if I've missed anything :smiley:

File: maintainers/maintainer-list.nix
Comment (afh) [inline maintainers/maintainer-list.nix:14482]: It seems that these are accidental changes, can you look into this, @savedra1?

**Review by afh — CHANGES_REQUESTED** [review]
Additionally to the comment below at least one file seems to have trailing blank spaces, which causes a test to fail. Could you address this please, @savedra1? 🙂

File: pkgs/by-name/cl/clipse/package.nix
Comment (afh) [inline pkgs/by-name/cl/clipse/package.nix:22]: According to the upstream repo's [`LICENCE`](https://github.com/savedra1/clipse/blob/main/LICENSE) the code is licensed under MIT.
```suggestion
    license = licenses.mit;
```

**Review by afh — CHANGES_REQUESTED** [review]
It seems there is also a mistake with the license declaration…

File: maintainers/maintainer-list.nix
Comment (savedra1) [reply maintainers/maintainer-list.nix]: Apologies! Have updated this now.

File: pkgs/by-name/cl/clipse/package.nix
Comment (savedra1) [reply pkgs/by-name/cl/clipse/package.nix]: Thanks again! this is now updated also.

**Comment by afh** [comment]
The [Checking EditorConfig / tests (pull_request_target)](https://github.com/NixOS/nixpkgs/actions/runs/8171665286/job/22340391010?pr=293722) still fails due to:
```
maintainers/maintainer-list.nix:
	Wrong line endings or new final newline
pkgs/by-name/cl/clipse/package.nix:
	Wrong line endings or new final newline
```

**Comment by savedra1** [comment]
> The [Checking EditorConfig / tests (pull_request_target)](https://github.com/NixOS/nixpkgs/actions/runs/8171665286/job/22340391010?pr=293722) still fails due to:
> 
> ```
> maintainers/maintainer-list.nix:
> 	Wrong line endings or new final newline
> pkgs/by-name/cl/clipse/package.nix:
> 	Wrong line endings or new final newline
> ```


@afh I appreciate your patience, my editor was behaving weirdly but should be all sorted now 😎 ❄️

File: pkgs/by-name/cl/clipse/package.nix
Comment (afh) [reply pkgs/by-name/cl/clipse/package.nix:24]: The quotes are no longer necessary now that your GitHub name is added to maintainers-list.nix
```suggestion
    maintainers = with maintainers; [ savedra1 ];
```

**Review by afh — CHANGES_REQUESTED** [review]
Thank you for your patience and effort addressing the PR comments,  @savedra1. Now that the files are almost in order (see comment below) let's start to get the commit history straight:

1. Please change the commit message on 8cd8b51cab914eec2154a3fa3296b5db98a290b9 to say `maintainers: add savedra1`
2. Squash all other commits into 10c06e26a1bffaf08c517e8fc5e4b3c693352b74
3. For extra bonus points swap the two PRs, so that the maintainers change comes before added package 😃

**Comment by afh** [comment]
Almost there, @savedra1, the maintainers commit still contains changes to `clipse/package.nix` that should be in the init commit…

**Comment by savedra1** [comment]
> Almost there, @savedra1, the maintainers commit still contains changes to `clipse/package.nix` that should be in the init commit…

@afh how are we looking?

**Comment by afh** [comment]
Result of `nixpkgs-review pr 293722` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>clipse</li>
  </ul>
</details>

**Comment by savedra1** [comment]
> Result of `nixpkgs-review pr 293722` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
> 
> 1 package failed to build:
> The changes look good to me, well done @savedra1. Unfortunately `nixpkgs-review` failed for me with the following error:
> 
> ```
> error: hash mismatch in fixed-output derivation '/nix/store/4ifzvjpy95bw8j18d8l8xj57k9d4y5wr-source.drv':
>          specified: sha256-zCsSfnZbhOYe1ysnGFO6VYZKVYCNZY0DvP2FB3EXFDM=
>             got:    sha256-DLvYTPlLkp98zCzmbeL68B7mHl7RY3ee9rL30vYm5Ow=
> error: 1 dependencies of derivation '/nix/store/9cbnn63panpmv6hy9648w4xk5794gmwn-clipse-0.0.6.drv' failed to build
> error: 1 dependencies of derivation '/nix/store/i8svp3i5mi8835hzczrdy2v82rpzd9dq-review-shell.drv' failed to build
> ```
> 
> Could you look into the hash mismatch?

Interesting, not sure why that hash stopped working as no changes were made to the upstream but have updated the new one - it built for me locally so hopefully that does the trick :crossed_fingers: @afh

**Comment by afh** [comment]
Result of `nixpkgs-review pr 293722` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>clipse</li>
  </ul>
</details>

**Review by afh — APPROVED** [review]
Thank you for your persistence and responsiveness on this PR, @savedra1, much appreciated! The changes look good to me now 👍

**Comment by savedra1** [comment]
thanks for all your help! @afh :heart: :snowflake:

File: pkgs/by-name/cl/clipse/package.nix
Comment (drupol) [inline pkgs/by-name/cl/clipse/package.nix:19]: Please, remove `with lib;`, see: https://github.com/NixOS/nixpkgs/pull/293767

File: pkgs/by-name/cl/clipse/package.nix
Comment (drupol) [inline pkgs/by-name/cl/clipse/package.nix:26]: Please sort meta attributes alphabetically.

**Review by drupol — CHANGES_REQUESTED** [review]
Hi !

Thanks for your contribution.

I've left some comments, let me know if this is clear enough.

File: pkgs/by-name/cl/clipse/package.nix
Comment (afh) [inline pkgs/by-name/cl/clipse/package.nix]:
```suggestion
    maintainers = [ lib.maintainers.savedra1 ];
```

**Review by afh — COMMENTED** [review]
See suggestion below :)

**Comment by savedra1** [comment]
Hi @drupol I think I understand. I have reformatted this but noticed when testing locally that the maintainer needs to be a string when using this forma. Is that expected or am I missing something?

**Review by drupol — CHANGES_REQUESTED** [review]
The PR LGTM, could you please make sure it only contains 2 commits.
1. One to add you as maintainer
2. One to introduce clipse

**Comment by savedra1** [comment]
Hi @drupol, hopefully this looks alright now? Thanks

**Review by drupol — APPROVED** [review]
Thank you ! Waiting for the CI to be done and merging this.

File: pkgs/by-name/cl/clipse/package.nix
Comment (SuperSandro2000) [inline pkgs/by-name/cl/clipse/package.nix:25]:
```suggestion
```
that's the default for buildGoModule

File: pkgs/by-name/cl/clipse/package.nix
Comment (SuperSandro2000) [inline pkgs/by-name/cl/clipse/package.nix:20]:
```suggestion
    description = "Useful clipboard manager TUI for Unix";
```
see contributing guide

File: pkgs/by-name/cl/clipse/package.nix
Comment (afh) [reply pkgs/by-name/cl/clipse/package.nix:25]: I wasn't aware of that, thanks for pointing this out, @SuperSandro2000 and apologies to you, @savedra1, for having created extra work for you.

File: pkgs/by-name/cl/clipse/package.nix
Comment (afh) [reply pkgs/by-name/cl/clipse/package.nix:20]: For my own future reference see section [**Meta attributes** in `pkgs/README.md`](https://github.com/NixOS/nixpkgs/tree/master/pkgs#meta-attributes)

Comment (savedra1) [reply pkgs/by-name/cl/clipse/package.nix]: Thanks both - have updated your suggestions now @SuperSandro2000 :+1:

**Comment by savedra1** [comment]
Hey @drupol just a friendly bump to see if this can be merged now? Thanks :)

**Comment by drupol** [comment]
Thx for the PR and for reminding me!

**Comment by savedra1** [comment]
Hi @drupol @afh I just have a question if that is ok. I can see the package has now been added to the Nixpkgs repo under the unstable section, but this fails to install for me when using the suggested methods such as `nix-shell -p clipse` or adding the package to my configuration.nix file. The errors suggest clipse is undefined/not found. Is this expected while on the unstable/master branch or do I need to backport this PR to the release branch in order to resolve? I have read through the docs but still a bit unsure. Thanks!

**Comment by drupol** [comment]
First thing is to look at https://nixpk.gs/pr-tracker.html?pr=293722
I wish we could have something similar integrated in Nix, but it's not yet the case.

Since you're using `nix-shell`, I guess you're using Nix channels right ?
Are they up-to-date?
Did you run `nix-channel --update` before?

If you're using flake, you can do: `nix shell github:NixOS/nixpkgs#clipse` (that works for me!)

I invite you to read this page: https://nixos.wiki/wiki/Nix_channels it might help people understand what's happening after a PR is merged. FYI, this is also sometimes complicated for me since most of the process are not correctly documented.

**Comment by savedra1** [comment]
Thanks for getting back to me @drupol :) the `nix shell github:NixOS/nixpkgs#clipse` works for me too using `nix shell`, but still unsure why the referenced methods of `nix-env -iA nixos...` and `environment.systemPackages = [ pkgs.hydra-check ];` does not work for `clipse` when it does work for other nix packages.. 

Thanks also for the documentation you shared just now. I installed `hydra-check` by adding the package to my **systemPackages** array, however when i try to do this with `clipse` I get an error __"error: undefined variable 'clipse'"__ 

I tested with the other new package that @afh referenced in the PR, `kyverno-chainsaw`, and this has the same issue.. As this is also only on the unstable branch I was assuming it's related to that but yeah i can't see anything else in the docs about this. 

I also have my flake set up to follow the unstable branch already for nixpgs. :thinking: 

Let me know if you have any other thoughts, appreciate you taking the time to look into it.

**Comment by savedra1** [comment]
Found the Hydra build info [here](https://hydra.nixos.org/build/252991633#tabs-summary) that suggests installation using `nix-env -i /nix/store/gi1li356jwr1h2nx67hf14vly3gz1wi2-clipse-0.0.6 --option binary-caches https://cache.nixos.org` which does work on my machine, can't see how to build this into my configuration.nix file though still.

**Pull request merged**
